python/examples/feature-selection.py

```diff
@@ -0,0 +1,116 @@
+# the names of the labels must be a subset of the names of the
+# properties produced by the calculator. You can see the default set
+# of names with:
+print("properties names:", descriptor.property_names)
+
+# Now we can use these names for selection.
+# In this case, only features  matching the labels in this selection
+# will be used
+selection = Labels(
+    names=["l"],
+    values=np.array([[0]]),
+)
+reference_descriptor = calculator.compute(frames, selected_properties=selection)
+
+# The previous selection method uses the same selection for all blocks.
+# If you can to use different selection for different blocks, you should
+# use a `TensorMap` to create your selection
+
+descriptor_for_comparison = calculator.compute(
+    frames, selected_properties=reference_descriptor
+)
+print("shape of first block initially:", {descriptor.block(0).values.shape})
+print(
+    "shape of first block of reference:", {reference_descriptor.block(0).values.shape}
+)
+print(
+    "shape of first block after selection:",
+    {descriptor_for_comparison.block(0).values.shape},
+)
+
+
+# The `TensorMap` format allows us to select features within
+# each block, and then construct a general matrix of features. We
+# can select the most significant features using FPS, which selects
+# features based on the  distance between them. The following algorithm
+# selects the 10 most important features in each block, then constructs
+# a new descriptor based on them, and calculates the final matrix of
+# features for it.
+
+# choose features:
+def feature_selection(descriptor_data, n_to_select):
+    features_list = []
+    for i in range(len(descriptor_data)):
+        fps = FPS(n_to_select=n_to_select)
+        values = descriptor_data.block(i).values
+        mask = fps.fit(values).get_support()
+        features_list.append(np.where(mask))
+    return features_list
+
+
+features_list = feature_selection(descriptor_data=descriptor, n_to_select=10)
+
+
+# create a new `TensorMap`
+set_of_blocks = []
+for i in range(len(features_list)):
+    indices = np.array(features_list[i]).reshape(
+        -1,
+    )
+    values = np.asarray(descriptor.block(i).values[:, indices])
+    components = descriptor.block(i).components
+    samples = descriptor.block(i).samples
+    property_values = []
+    for j in range(len(indices)):
+        property_values.append(
+            [
+                descriptor.block(i).properties[j][0],
+                descriptor.block(i).properties[j][1],
+                descriptor.block(i).properties[j][2],
+            ]
+        )
+    property_values = np.asarray(property_values).reshape(-1, 3)
+    properties = Labels(descriptor.block(i).properties.names, property_values)
+    set_of_blocks.append(
+        TensorBlock(
+            values=values, samples=samples, components=components, properties=properties
+        )
+    )
+new_descriptor = TensorMap(descriptor.keys, set_of_blocks)
```

Comment:
This is manually re-doing what rascaline can do for you. I would do something like
```suggestion
# choose features:
def fps_feature_selection(descriptor, n_to_select):
    blocks = []
    for _, block in descriptor:
        # create a separate FPS selector for each block
        fps = FPS(n_to_select=n_to_select)
        mask = fps.fit(block.values).get_support()
        selected_properties = block.properties[mask]

        # put the selected features in a format rascaline can use
        blocks.append(TensorBlock(
            # values, samples and component carry no information here
            values=np.empty(),
            samples=Labels.single(),
            components=[],
            properties=selected_properties
        ))

        
    return TensorMap(descriptor.keys, blocks)


selection = fps_feature_selection(descriptor, n_to_select=10)

# use the selection with rascaline
selected_descriptor = calculator.compute(frames, selected_properties=selection)
```